main.py

```diff
@@ -430,17 +427,19 @@ def delete_circuit(self, request: Request) -> JSONResponse:
             result = f"circuit_id {circuit_id} not found"
             log.debug("delete_circuit result %s %s", result, 404)
             raise HTTPException(404, detail=result) from KeyError
-
         log.info("Removing %s", evc)
+
         with evc.lock:
-            evc.remove_current_flows()
-            evc.remove_failover_flows(sync=False)
-            evc.deactivate()
-            evc.disable()
-            self.sched.remove(evc)
-            evc.archive()
-            evc.remove_uni_tags()
-            evc.sync()
+            if not evc.archived:
```

Comment:
Great guard here, this solves EVC operations thread safety per EVC, but notice that it's still possible that the `"deleted"` event gets sent twice, which would be redundant. If you bring `emit_event` to this with block here it'd solve this last remainig part, see what I mean?

Reply:
Moved `emit_event`

Reply:
Running e2e tests